class-wc-calypso-bridge-shared.php

```diff
@@ -123,6 +124,13 @@ public function add_scripts() {
 			'version'                      => WC_CALYPSO_BRIDGE_CURRENT_VERSION,
 		);
 
+		if ( $is_ecom_trial ) {
+			$introductory_offers = WC_Calypso_Bridge_Introductory_offers::get_current_offers_from_current_blog();
+			if ( count( $introductory_offers ) ) {
+				$params['introductoryOffer'] = WC_Calypso_Bridge_Introductory_offers::extract_offer_data_for_js( current( $introductory_offers ) );
```

Comment:
I wasn't sure if we have a logic to select an offer when there are more than 1. For now, I'm choosing the first one with current(). @ilyasfoo

Reply:
Hmm, I think it's better to just populate the whole array into the parameter, then our frontend can decide how to manage it. 

Might be better since we're preserving `introductoryOffer`'s type (array instead of a single object) in case of a future change. What do you think?

Reply:
That's a good for future proofing, but we also need to consider the data size. We're loading it for every page load. Since this is pretty easy to change, I think we should expose the whole array when we actually have a use case. What do you think?

Reply:
Do you happen to have a link to the source code that explains how Calypso selects which plan to display when there are multiple offers?

Reply:
Good point, I agree that producing all offers is premature 👍 

> Do you happen to have a link to the source code that explains how Calypso selects which plan to display when there are multiple offers (Does it even have a logic for selecting only one 🤔 )?

IMO calypso don't have the same problem since the app is JS and no server is involved, so everything is already available regardless if they only need 1 plan or more.

Reply:
@ilyasfoo I'm not clear on whether we can have multiple offers with different terms while using only one banner. Does this mean we have to choose just one offer, perhaps the first one? 🤔

Reply:
@moon0326 Gotcha. Since the introductory offer is the same for all Woo Express plans, I think this is how we should handle it:

1. Perhaps instead of `introductoryOffer`, let's rename it to `wooExpressIntroductoryOffer`
1. When deciding which plan to take, loop and check if it's any of [Woo Express plan slugs](https://github.com/Automattic/wp-calypso/blob/63c08c6989595742db1f9aaf70216663198db29b/packages/calypso-products/src/constants/wpcom.ts#L39-L42) and return the first one found

Reply:
Updated in https://github.com/Automattic/wc-calypso-bridge/pull/1397/commits/f2f26ed8676527a782671bf5352fe97ef4ecce68